src/dos/dos.cpp

```diff
@@ -1204,6 +1204,25 @@ static Bitu DOS_26Handler(void) {
     return CBRET_NONE;
 }
 
+void set_ver(char *args, bool start)
+{
+	char *word = StripWord(args);
+	if (!*args && !*word) { // Reset
+		dos.version.major = 5;
+		dos.version.minor = 0;
+	} else if (*args == 0 && *word && (strchr(word, '.') != 0)) { // Allow:
+		                                                      // ver set 5.1
+		const char *p = strchr(word, '.');
+		dos.version.major = (Bit8u)(atoi(word));
+		dos.version.minor = (Bit8u)(strlen(p + 1) == 1 && *(p + 1) > '0' &&
+		                                            *(p + 1) <= '9'
+		                                    ? atoi(p + 1) * 10
+		                                    : atoi(p + 1));
+	} else { // Official syntax: ver set 5 2
+		dos.version.major = (Bit8u)(atoi(word));
+		dos.version.minor = (Bit8u)(atoi(args));
+	}
```

Comment:
![2020-07-23_12-22](https://user-images.githubusercontent.com/1557255/88330221-f21d4800-ccdf-11ea-876f-c751f663b15e.png)


If the users feeds it broken values, suggest:
1. Informing them
2. Falling back to the default version (5.0?)

If the user feeds it a version outside the acceptable range, suggest clamping the value and also informing the user.  For example, the lowest DOS version I could find was 1.24 and the greatest version was 7.1.

https://en.wikipedia.org/wiki/MS-DOS#Versions

Reply:
I think it is possible to do a sanity check, but we need to think of a good range for this. If we only consider MS-DOS or PC DOS, the first official version was 1.0 (PC DOS) and the last version was 8.0 (MS-DOS, included in Windows ME with real-mode support removed). But if we consider other DOS too, then there are even more versions. For example, the earliest version of FreeDOS was 0.01 (alpha), and the last version of DR-DOS was 8.1. Furthermore, there are also other emulated DOS version numbers, for example, I have heard that OS/2 reports a DOS version greater than 10. So perhaps it is fine to accept a version in the range like 0-100.

Reply:
Yeah, sounds reasonable to push the bounds out a bit. Maybe 1.00 to 15 (0xf)?
I couldn't imagine DOS software refusing to run on 7.x and instead needing 8.x (curious if there are any examples out there).
@dreamer, what are your thoughts on sane boundaries?

Reply:
@kcgen DOS programs included with Windows ME (ATTRIB, FIND, etc) will show the error "Incorrect MS-DOS version" unless the reported DOS version is 8.0. Also, if you try to run its WIN.COM with a reported version of 7.x or lower, it will show the following error:

``This version of Windows runs only on MS-DOS 8.00.``

But of course these are the only examples of DOS software refusing to run on 7.x and instead needing 8.x.

Reply:
Wow; interesting! Thanks @Wengier.
Maybe 15 is still a sane upper-bound? Perhaps we go a bit tighter - to say 11 or 12?.

Reply:
@kcgen The problem with numbers like 11, 12 and 15 are kind of arbitrary. Perhaps need to explain why they are chosen as the upper-bound.

Reply:
Fully agree; I'm just trying to tighten them down to a "reasonable" conservative boundary -- you have examples of software needing 8.x (so that's a hard boundary), and mentioned OS/2 reporting > 10.x or 11.x? But no examples of software requiring those versions.  So I think stretching a bit beyond this is a safe but still generous boundary.

Reply:
@kcgen I just tried to install and start OS/2 Wrap 3.0 myself in a virtual machine, and can confirm that OS/2 Wrap 3.0 reports "20.30" as the default DOS version number. Screenshot:

![OS_2-2020-07-24-15-35-38](https://user-images.githubusercontent.com/8216923/88428954-d1c6b980-cdc3-11ea-8bbe-5a5b8d89f20f.png)

Reply:
@kcgen Also, I can confirm that DOS tools from OS/2 Wrap 3.0 (such as assign.com) can run in DOS(Box) if the reported DOS version is set to 20.30.

Reply:
According to https://dos4gw.org/213306_-_DOS_5_0_-_GET_TRUE_VERSION_NUMBER the upper bound is for minor version is 99; that site is generally quite interesting for investigating various DOS related details :)

Reply:
@dreamer, neat reference! We need a list of these sites in the wiki :-)  

Given the new details, would 30.99 be a sane upper bound?

Reply:
The upper bound is now 30.99 as mentioned (30 for major and 99 for minor).